Userland/Libraries/LibAudio/UserSampleQueue.cpp

```diff
@@ -51,6 +51,8 @@ size_t UserSampleQueue::size()
 size_t UserSampleQueue::remaining_samples()
 {
     Threading::MutexLocker lock(m_sample_mutex);
+    if (m_backing_samples.size() <= m_samples_to_discard)
+        return 0;
```

Comment:
The UserSampleQueue is a AudioServerConnection-internal utility class, it only has its own file for convenience. It is not intended to be used outside (it also has very little use outside this specific purpose) and misusing the class is always allowed to trigger a VERIFY. For example, Vector triggers a VERIFY on wrong indices, even though there's a lot of varied users.

Reply:
Sure, that's fine, let me push an update.